bindings/java/src/main/java/org/ucx/jucx/ucp/UcpWorkerParams.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (C) Mellanox Technologies Ltd. 2019. ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+
+package org.ucx.jucx.ucp;
+
+import java.nio.ByteBuffer;
+import java.util.BitSet;
+
+import org.ucx.jucx.ucs.UcsConstatns;
+
+public class UcpWorkerParams {
+
+    /**
+     * Mask of valid fields in this structure.
+     * Fields not specified in this mask would be ignored.
+     * Provides ABI compatibility with respect to adding new fields.
+     */
+    private long fieldMask;
+
+
+    private int threadMode;
+
+    private BitSet cpuMask = new BitSet();
+
+    private long events;
+
+    private ByteBuffer userData;
+
+    private int eventFD;
+
+    /**
+     * Suggests the thread safety mode which workerand the associated resources
+     * should be created with. The default value is UCS_THREAD_MODE_SINGLE and
+     * it is used when the value of the parameter is not set. When this
+     * parameter is set, the {@link UcpWorker#UcpWorker(UcpContext)}
+     * attempts to create worker with this thread mode.
+     * The thread mode with which worker is created can differ from the
+     * suggested mode.
+     */
+    public UcpWorkerParams setThreadMode(UcsConstatns.UcsThreadMode threadMode) {
+        this.fieldMask |= UcpConstants.UCP_WORKER_PARAM_FIELD_THREAD_MODE;
+        this.threadMode = threadMode.ordinal();
+        return this;
+    }
+
+    /**
+     * Mask of which CPUs worker resources should preferably be allocated on.
+     * This value is optional.
+     * If it's not set, resources are allocated according to system's default policy.
+     */
+    public UcpWorkerParams setCpu(int cpuNum) {
+        this.fieldMask |=  UcpConstants.UCP_WORKER_PARAM_FIELD_CPU_MASK;
+        this.cpuMask.set(cpuNum);
+        return this;
+    }
+
+    /**
+     * Remote memory access send completion.
+     */
+    public UcpWorkerParams requestWakeupRMA() {
```

Comment:
IMO, all requestXXX() func should be setters, with void return type and ```boolean enable``` arg to fit java semantic (same for getters), i.e.
```
public void setWakeupRMA(boolean enable)
{
     if (enable) {
     } else {
     }
}
```

Reply:
Another option might be to add `void reset()` or `void clear()` method to make the parameters object reusable for building many workers with different parameters.